subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationConfiguration.kt

```diff
@@ -42,44 +40,66 @@ open class InstrumentationConfiguration(val name: String) {
      */
     var reportSkippedTests = false
 
-    var annotatedWith: Collection<String>? = null
+    /**
+     * Applied only in next instrumentation invoke at the same git HEAD
+     * It remove already succeed test from execution
+     */
+    @Deprecated("since 2020.3.6")
+    var rerunFailedTests = true
 
-    var impactAnalysisPolicy: ImpactAnalysisPolicy = ImpactAnalysisPolicy.Off
+    @Deprecated("since 2020.3.6")
+    var annotatedWith: Collection<String>? = null
 
+    @Deprecated("since 2020.3.6")
     var tests: List<String>? = null
 
+    @Deprecated("since 2020.3.6")
     var prefixFilter: String? = null
 
-    var kubernetesNamespace = "android-emulator"
-
-    var enableDeviceDebug: Boolean = false
-
     /**
      * It must be a valid reportId.
      * Get test runs from report by id. Filter already succeed or new tests.
      * Stay only failed contained in report
      */
+    @Deprecated("since 2020.3.6")
     var keepFailedTestsFromReport: String? = null
 
-    lateinit var targetsContainer: NamedDomainObjectContainer<TargetConfiguration>
+    var impactAnalysisPolicy: ImpactAnalysisPolicy = ImpactAnalysisPolicy.Off
+
+    var kubernetesNamespace = "android-emulator"
+
+    var enableDeviceDebug: Boolean = false
+
+    var filter = "default"
+
+    abstract val targetsContainer: NamedDomainObjectContainer<TargetConfiguration>
 
     val targets: List<TargetConfiguration>
         get() = targetsContainer.toList()
             .filter { it.enabled }
 
+    @Deprecated("since 2020.3.6 https://docs.gradle.org/current/userguide/kotlin_dsl.html#sec:interoperability", replaceWith = ReplaceWith("targets(Action<NamedDomainObjectContainer<TargetConfiguration>>)"))
```

Comment:
The interoperability guide says that we need to use `Action<T>`. Gradle automatically converts groovy closures and kotlin lambda to it.